src/metadata/EventMetadata.ts

```diff
@@ -0,0 +1,33 @@
+import type { EventId, IEvent } from '../ISequentialEventListener';
+import 'reflect-metadata';
+
+export const SEQUENTIAL_EVENT = '__sequentialEvent';
+
+export class EventMetadata {
+  constructor(private readonly event: FunctionConstructor) {
+  }
+
+  static from(eventOrConstructor: FunctionConstructor | IEvent) {
+    if (EventMetadata.isEventConstructor(eventOrConstructor)) {
+      return new EventMetadata(eventOrConstructor);
+    } else {
+      return new EventMetadata(Object.getPrototypeOf(eventOrConstructor).constructor);
+    }
+  }
+
+  public getEventId(): EventId {
+    const eventMetadata = Reflect.getMetadata(SEQUENTIAL_EVENT, this.event);
+
+    return eventMetadata?.id;
+  }
+
+  public assignId(id: EventId) {
+    if (!Reflect.hasOwnMetadata(SEQUENTIAL_EVENT, this.event)) {
```

Comment:
What happens if I add the same event to two different @SequentialEventListener?

As far as I can tell then id will be assigned only once.
Is it correct?

Reply:
Yes, it will work as expected. We have the `Map<EventId, Listener[]>` to find listeners corresponding to specific event ID.